.gitignore

```diff
@@ -2,3 +2,4 @@
 **/target
 **/result
 **/*.swp
+**/.idea
```

Comment:
This is new since the first review. I switched back to CLion and IntelliJ for Rust dev 😅

Reply:
i'm fine with having editor-specific gitignore configurations, i think, but i do have a preference for merging this kind of change separately, so that it's not part of the git history for the unrelated documentation changes in this PR?